src/MahApps.Metro/MahApps.Metro.Shared/Controls/Helper/TextBoxHelper.cs

```diff
@@ -66,6 +85,61 @@ public static void SetIsSpellCheckContextMenuEnabled(UIElement element, bool val
             element.SetValue(IsSpellCheckContextMenuEnabledProperty, value);
         }
 
+#if NET4_5
+        [Category(AppName.MahApps)]
+        [AttachedPropertyBrowsableForType(typeof(TextBox))]
+        [AttachedPropertyBrowsableForType(typeof(ComboBox))]
+        [AttachedPropertyBrowsableForType(typeof(DatePicker))]
+        [AttachedPropertyBrowsableForType(typeof(TimePickerBase))]
+        [AttachedPropertyBrowsableForType(typeof(NumericUpDown))]
+        public static bool GetAutoWatermark(DependencyObject element)
+        {
+            return (bool)element.GetValue(AutoWatermarkProperty);
+        }
+
+        public static void SetAutoWatermark(DependencyObject element, bool value)
+        {
+            element.SetValue(AutoWatermarkProperty, value);
+        }
+
+        private static void OnAutoWatermarkChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            FrameworkElement element = d as FrameworkElement;
+            if (element != null)
+            {
+                element.Loaded += OnControlWithAutoWatermarkSupportLoaded;
```

Comment:
Because of the event subscription? Shouldn't it be enough to unsubscribe directly after the event was raised the first time?